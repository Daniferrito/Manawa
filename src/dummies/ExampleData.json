{
    "categories": [
        {
            "title": "Work",
            "tasks": [
                {
                    "title": "Write report",
                    "description": "This is a description",
                    "expirationDate": "2020-01-20T12:00:00Z"
                },
                {
                    "title": "Meeting with stakeholders",
                    "description": "This is a description",
                    "expirationDate": "2020-01-21T12:00:00Z"
                }
            ]
        },
        {
            "title": "Personal",
            "tasks": [
                {
                    "title": "Buy food",
                    "description": "Fill the fridge",
                    "expirationDate": "2020-01-24T18:00:00Z"
                }
            ]
        }
    ]
}