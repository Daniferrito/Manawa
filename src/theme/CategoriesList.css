.container{
    display: grid;
    grid-template-columns: 50% 50%;
}

.title{
    grid-column-start: 1;
    grid-column-end: -1;
}

.subElement{
    grid-column: auto;
}