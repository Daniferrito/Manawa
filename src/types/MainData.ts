import Category from './Category';

export default class MainData {
  categories: Category[];
  constructor(categories: Category[]) {
    this.categories = categories;
  }
  static fromJson(json: any): MainData {
    return new MainData(json.categories.map((category: any) => Category.fromJson(category)));
  }
}
