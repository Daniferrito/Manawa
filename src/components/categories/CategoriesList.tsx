import React from 'react';
import Category from '../../types/Category';
import CategoriesListItem from './CategoriesListItem';

import ExampleData from '../../dummies/ExampleData.json';
import MainData from '../../types/MainData';

interface Props {
  categories?: Category[];
}

const CategoriesList = (props: Props) => {
  const { categories = MainData.fromJson(ExampleData).categories } = props;

  // if (categories === undefined) return <>{'Error'}</>;

  return (
    <>
      {categories.map(category => (
        <CategoriesListItem key={category.title} category={category} />
      ))}
    </>
  );
};

export default CategoriesList;
