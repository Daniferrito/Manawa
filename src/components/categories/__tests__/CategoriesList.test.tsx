import React from 'react';
import { shallow } from 'enzyme';
import CategoriesList from '../CategoriesList';

import ExampleData from '../../../dummies/ExampleData.json';
import MainData from '../../../types/MainData';

test('renders without crashing', () => {
  const categoriesList = shallow(
    <CategoriesList categories={MainData.fromJson(ExampleData).categories} />
  );
  expect(categoriesList).toMatchSnapshot();
});
test('renders without crashing2', () => {
  const categoriesList = shallow(<CategoriesList />);
  expect(categoriesList).toMatchSnapshot();
});
