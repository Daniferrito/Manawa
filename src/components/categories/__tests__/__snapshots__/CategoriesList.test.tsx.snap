// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Fragment>
  <CategoriesListItem
    category={
      Category {
        "tasks": Array [
          Task {
            "description": "This is a description",
            "expirationDate": 2020-01-20T12:00:00.000Z,
            "title": "Write report",
          },
          Task {
            "description": "This is a description",
            "expirationDate": 2020-01-21T12:00:00.000Z,
            "title": "Meeting with stakeholders",
          },
        ],
        "title": "Work",
      }
    }
    key="Work"
  />
  <CategoriesListItem
    category={
      Category {
        "tasks": Array [
          Task {
            "description": "Fill the fridge",
            "expirationDate": 2020-01-24T18:00:00.000Z,
            "title": "Buy food",
          },
        ],
        "title": "Personal",
      }
    }
    key="Personal"
  />
</Fragment>
`;

exports[`renders without crashing2 1`] = `
<Fragment>
  <CategoriesListItem
    category={
      Category {
        "tasks": Array [
          Task {
            "description": "This is a description",
            "expirationDate": 2020-01-20T12:00:00.000Z,
            "title": "Write report",
          },
          Task {
            "description": "This is a description",
            "expirationDate": 2020-01-21T12:00:00.000Z,
            "title": "Meeting with stakeholders",
          },
        ],
        "title": "Work",
      }
    }
    key="Work"
  />
  <CategoriesListItem
    category={
      Category {
        "tasks": Array [
          Task {
            "description": "Fill the fridge",
            "expirationDate": 2020-01-24T18:00:00.000Z,
            "title": "Buy food",
          },
        ],
        "title": "Personal",
      }
    }
    key="Personal"
  />
</Fragment>
`;
